nemo/collections/tts/torch/data_ssl_vocoder.py

```diff
@@ -0,0 +1,590 @@
+                return_complex=True,
+                center=True,
+            )
+
+            if spec.dtype in [torch.cfloat, torch.cdouble]:
+                spec = torch.view_as_real(spec)
+            spec = torch.sqrt(spec.pow(2).sum(-1) + EPSILON)
+
+            mel = torch.matmul(self.fb.to(spec.dtype), spec)
+            log_mel = torch.log(torch.clamp(mel, min=torch.finfo(mel.dtype).tiny))[0]
+
+            torch.save(log_mel, mel_spec_fp)
+
+            return log_mel
+
+    def segment_wav(self, wav, segment_length=44100, hop_size=22050, min_segment_size=22050):
+        if len(wav) < segment_length:
+            pad = torch.zeros(segment_length - len(wav))
+            segment = torch.cat([wav, pad])
+            return [segment]
+        else:
+            si = 0
+            segments = []
+            while si < len(wav) - min_segment_size:
+                segment = wav[si : si + segment_length]
+                if len(segment) < segment_length:
+                    pad = torch.zeros(segment_length - len(segment))
+                    segment = torch.cat([segment, pad])
+                segments.append(segment)
+                si += hop_size
+            return segments
+
+    def get_ssl_features(self, audio_ssl, audio_ssl_length, wav_text_id):
+        content_emb_fn = f"{self.ssl_content_emb_type}_content_embedding_{wav_text_id}.pt"
+        speaker_emb_fn = f"speaker_embedding_{wav_text_id}.pt"
+        duration_fn = f"duration_embedding_{wav_text_id}.pt"  # embedding just for namesake
+        content_emb_fp = os.path.join(self.sup_data_dir, content_emb_fn)
+        speaker_emb_fp = os.path.join(self.sup_data_dir, speaker_emb_fn)
+        duration_fp = os.path.join(self.sup_data_dir, duration_fn)
+        if os.path.exists(content_emb_fp):
+            content_embedding = torch.load(content_emb_fp)
+            if os.path.exists(speaker_emb_fp):
+                speaker_embedding = torch.load(speaker_emb_fp)
+            else:
+                raise ValueError(f"Speaker embedding file {speaker_emb_fp} does not exist")
+            encoded_len = torch.tensor(content_embedding.shape[1]).long()
+            if os.path.exists(duration_fp):
+                duration = torch.load(duration_fp)
+            else:
+                # default duration is 4 frames per embedding
+                duration = torch.ones(content_embedding.shape[1]) * 4.0
+            return content_embedding, speaker_embedding, encoded_len, duration
+        else:
+            with torch.no_grad():
+                (
+                    _,
+                    speaker_embedding_normalized,
+                    content_embedding,
+                    content_log_probs,
+                    encoded_len,
+                ) = self.ssl_model.forward_for_export(
+                    input_signal=audio_ssl[None],
+                    input_signal_length=audio_ssl_length[None],
+                    normalize_content=self.normalize_content,
+                )
```

Comment:
Currently, it does require a lot of config parameters in the init function from the saved checkpoint. Will need to take a closer look to see if it can be made optional.